dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -114,21 +114,36 @@ void dpnp_rng_binomial_c(void* result, const int ntrial, const double p, const s
     }
     _DataType* result1 = reinterpret_cast<_DataType*>(result);
 
-    if (dpnp_queue_is_cpu_c())
+    if (ntrial == 0 || p == 0)
     {
-        mkl_rng::binomial<_DataType> distribution(ntrial, p);
-        // perform generation
-        auto event_out = mkl_rng::generate(distribution, DPNP_RNG_ENGINE, size, result1);
-        event_out.wait();
+        _DataType* fill_value = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(sizeof(_DataType)));
+        fill_value[0] = 0;
+        dpnp_initval_c<_DataType>(result, fill_value, size);
+    }
+    else if (p == 1)
+    {
+        _DataType* fill_value = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(sizeof(_DataType)));
+        fill_value[0] = static_cast<_DataType>(ntrial);
```

Comment:
I think we need to free `fill_value` after it's not used more.

Reply:
oh, yeah, of course. Thanks